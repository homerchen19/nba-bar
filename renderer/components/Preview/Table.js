import React from 'react';
import PropTypes from 'prop-types';
import R from 'ramda';
import { Flex } from 'antd-mobile';
import styled from 'styled-components';

import { colors } from '@styles/theme';

const Wrapper = styled.div`
  display: block;
  flex: 3;
  width: 100%;
  overflow-y: scroll !important;
  border-top: 1px solid ${colors.white};
  border-bottom: 1px solid ${colors.white};
  background: #fff;
`;

const Row = styled(Flex)`
  flex: 1;
  width: 100%;
  padding: 10px 15px;
  border-top: 1px solid ${colors.white};
  border-bottom: 1px solid ${colors.white};
`;

const Column = styled(Flex.Item)`
  margin: 0 !important;
  text-align: ${props => props.align};
`;

const Table = ({ home, visitor }) => (
  <Wrapper>
    {R.map(
      key => (
        <Row key={key} justify="center">
          <Column align="left">
            <p>{home[key]}</p>
          </Column>
          <Column align="center">
            <b>
              <p>
                {key === 'plus_minus'
                  ? '+/-'
                  : key.toUpperCase().replace('_PCT', '%')}
              </p>
            </b>
          </Column>
          <Column align="right">
            <p>{visitor[key]}</p>
          </Column>
        </Row>
      ),
      R.keys(home)
    )}
  </Wrapper>
);

Table.propTypes = {
  home: PropTypes.object.isRequired,
  visitor: PropTypes.object.isRequired,
};

export default Table;
