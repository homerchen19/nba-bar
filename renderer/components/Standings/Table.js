import React from 'react';
import PropTypes from 'prop-types';
import { Flex } from 'antd-mobile';
import { Table as VirtualizedTable } from 'react-virtualized';
import styled from 'styled-components';

import { colors } from '@styles/theme';
import nba from '@utils/nba';

const Wrapper = styled(Flex)`
  width: 100%;
  height: 100%;
  overflow-y: scroll !important;
`;

const HeaderCell = styled.div`
  width: ${props => props.width}px;
  padding: 8px 14px;
  background: ${colors.darkBlue};
  color: #fff;
  font-weight: 600;
  text-align: center;
`;

const Cell = styled.div`
  width: ${props => props.width}px;
  padding: 6px 14px;
  border-bottom: 1px solid ${colors.white};
  background: #fff;
  text-align: ${props => props.align || 'center'};
`;

const Conference = styled.h3`
  padding: 10px 0;
  background: ${colors.blue};
  color: #fff;
  text-align: center;
  text-transform: uppercase;
`;

const headerRowRenderer = ({ className, style }) => (
  <Flex className={className} style={style}>
    <HeaderCell key="Teams" width="75">
      Teams
    </HeaderCell>
    <HeaderCell key="W" width="47">
      W
    </HeaderCell>
    <HeaderCell key="L" width="47">
      L
    </HeaderCell>
    <HeaderCell key="WIN%" width="73">
      WIN%
    </HeaderCell>
    <HeaderCell key="GB" width="59">
      GB
    </HeaderCell>
  </Flex>
);

headerRowRenderer.propTypes = {
  className: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
};

const rowRenderer = ({ rowData, style }) => (
  <Flex key={rowData.teamId} style={style}>
    <Cell
      key="name"
      align="left"
      style={{
        color: '#fff',
        background: nba.getTeamBackgroundColor(rowData.name),
        fontWeight: '500',
      }}
      width="75"
    >
      {rowData.name}
    </Cell>
    <Cell key="W" width="47">
      {rowData.win}
    </Cell>
    <Cell key="L" width="47">
      {rowData.loss}
    </Cell>
    <Cell key="WIN %" width="73">
      {rowData.winPct}
    </Cell>
    <Cell key="GB" width="59">
      {rowData.gamesBehind}
    </Cell>
  </Flex>
);

rowRenderer.propTypes = {
  rowData: PropTypes.object.isRequired,
  style: PropTypes.object.isRequired,
};

const renderConferenceTable = (teams, conference) => (
  <div key={conference} style={{ width: '100%' }}>
    <Conference>{conference}</Conference>
    <VirtualizedTable
      width={300}
      height={300}
      headerHeight={32}
      rowHeight={30}
      rowCount={teams.length}
      rowGetter={({ index }) => teams[index]}
      headerRowRenderer={headerRowRenderer}
      rowRenderer={rowRenderer}
    />
  </div>
);

const Table = ({ east, west }) => (
  <Wrapper direction="column" align="center">
    {renderConferenceTable(east, 'east')}
    {renderConferenceTable(west, 'west')}
  </Wrapper>
);

Table.propTypes = {
  east: PropTypes.arrayOf(PropTypes.object).isRequired,
  west: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Table;
