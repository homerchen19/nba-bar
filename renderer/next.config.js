const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
