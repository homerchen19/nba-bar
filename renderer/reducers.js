import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
