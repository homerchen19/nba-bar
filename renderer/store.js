import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
