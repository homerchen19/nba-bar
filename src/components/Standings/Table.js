import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import R from 'ramda';
import { Flex } from 'antd-mobile';
import { StickyTable, Row, Cell } from 'react-sticky-table';
import styled from 'styled-components';
import { getMainColor } from 'nba-color';

import { colors } from '../../styles/theme';

const Wrapper = styled(Flex)`
  width: 100%;
  overflow-y: scroll !important;

  ::-webkit-scrollbar {
    display: none;
  }
`;

const Item = styled(Flex.Item)`
  width: 100%;
  margin: 0 !important;
`;

const HeaderCell = styled(Cell)`
  padding: 8px 14px;
  background: ${colors.darkBlue};
  color: #fff;
  font-weight: 600;
  text-align: center;
`;

const StyledCell = styled(Cell)`
  width: ${props => props.width}px;
  padding: 6px 14px;
  border-bottom: 1px solid ${colors.white};
  background: #fff;
  text-align: ${props => props.align || 'center'};
`;

const Conference = styled.h3`
  padding: 10px 0;
  background: ${colors.blue};
  color: #fff;
  text-align: center;
  text-transform: uppercase;
`;

const renderHeaderCell = () => {
  const cells = ['Teams', 'W', 'L', 'WIN%', 'GB'];

  return cells.map(cell => <HeaderCell key={cell}>{cell}</HeaderCell>);
};

const renderConferenceTable = (team, conference) => (
  <Fragment>
    <Conference>{conference}</Conference>
    <StickyTable key={conference} stickyColumnCount={0}>
      <Row key="header">{renderHeaderCell()}</Row>
      {R.map(
        teamData => (
          <Row key={teamData.teamId}>
            <StyledCell
              key="name"
              align="left"
              style={{
                color: '#fff',
                background: getMainColor(teamData.name).hex,
                fontWeight: '500',
              }}
              width="75"
            >
              {teamData.name}
            </StyledCell>
            <StyledCell key="W" width="47">
              {teamData.win}
            </StyledCell>
            <StyledCell key="L" width="47">
              {teamData.loss}
            </StyledCell>
            <StyledCell key="WIN %" width="73">
              {teamData.winPct}
            </StyledCell>
            <StyledCell key="GB" width="59">
              {teamData.gamesBehind}
            </StyledCell>
          </Row>
        ),
        team
      )}
    </StickyTable>
  </Fragment>
);

const Table = ({ east, west }) => (
  <Wrapper direction="column" align="center">
    <Item>{renderConferenceTable(east, 'east')}</Item>
    <Item>{renderConferenceTable(west, 'west')}</Item>
  </Wrapper>
);

Table.propTypes = {
  east: PropTypes.arrayOf(PropTypes.object).isRequired,
  west: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Table;
