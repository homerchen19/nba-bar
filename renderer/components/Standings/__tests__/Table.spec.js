import React from 'react';
import renderer from 'react-test-renderer';

import Table from '../Table';

const realMath = Object.create(global.Math);

beforeAll(() => {
  const mockMath = Object.create(global.Math);
  mockMath.random = () => 0.5;
  global.Math = mockMath;
});

afterAll(() => {
  global.Math = realMath;
});

it('be defined', () => {
  expect(Table).toBeDefined();
});

it('match snapshot', () => {
  const props = {
    east: [
      {
        gamesBehind: '0',
        loss: '22',
        name: 'TOR',
        teamId: '1610612761',
        win: '56',
        winPct: '.718',
      },
      {
        gamesBehind: '3',
        loss: '25',
        name: 'BOS',
        teamId: '1610612738',
        win: '53',
        winPct: '.679',
      },
      {
        gamesBehind: '7.5',
        loss: '30',
        name: 'CLE',
        teamId: '1610612739',
        win: '49',
        winPct: '.620',
      },
    ],
    west: [
      {
        gamesBehind: '0',
        loss: '15',
        name: 'HOU',
        teamId: '1610612745',
        win: '64',
        winPct: '.810',
      },
      {
        gamesBehind: '7',
        loss: '22',
        name: 'GSW',
        teamId: '1610612744',
        win: '57',
        winPct: '.722',
      },
      {
        gamesBehind: '16',
        loss: '31',
        name: 'POR',
        teamId: '1610612757',
        win: '48',
        winPct: '.608',
      },
    ],
  };
  const tree = renderer.create(<Table {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});
