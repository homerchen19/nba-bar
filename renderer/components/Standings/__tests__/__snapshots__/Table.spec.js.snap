// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="am-flexbox Table__Wrapper-sc-9u2sr5-0 hbRiRB am-flexbox-dir-column am-flexbox-align-center"
>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h3
      className="Table__Conference-sc-9u2sr5-3 iKGnKJ"
    >
      east
    </h3>
    <div
      className="ReactVirtualized__Table"
      role="grid"
      style={Object {}}
    >
      <div
        className="am-flexbox ReactVirtualized__Table__headerRow am-flexbox-align-center"
        style={
          Object {
            "height": 32,
            "overflow": "hidden",
            "paddingRight": 0,
            "width": 300,
          }
        }
      >
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 jhHCnp"
          width="75"
        >
          Teams
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 iGaVpi"
          width="47"
        >
          W
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 iGaVpi"
          width="47"
        >
          L
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 eZeJyr"
          width="73"
        >
          WIN%
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 cBrKBo"
          width="59"
        >
          GB
        </div>
      </div>
      <div
        aria-label="grid"
        aria-readonly={true}
        className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
        onScroll={[Function]}
        role="rowgroup"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "boxSizing": "border-box",
            "direction": "ltr",
            "height": 268,
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "relative",
            "width": 300,
            "willChange": "transform",
          }
        }
        tabIndex={0}
      >
        <div
          className="ReactVirtualized__Grid__innerScrollContainer"
          role="rowgroup"
          style={
            Object {
              "height": 90,
              "maxHeight": 90,
              "maxWidth": 300,
              "overflow": "hidden",
              "pointerEvents": "",
              "position": "relative",
              "width": "auto",
            }
          }
        >
          <div
            className="am-flexbox am-flexbox-align-center"
            style={
              Object {
                "height": 30,
                "left": 0,
                "overflow": "hidden",
                "paddingRight": 0,
                "position": "absolute",
                "top": 0,
                "width": 300,
              }
            }
          >
            <div
              className="Table__Cell-sc-9u2sr5-2 fjWUKQ"
              style={
                Object {
                  "background": "#ce1141",
                  "color": "#fff",
                  "fontWeight": "500",
                }
              }
              width="75"
            >
              TOR
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              56
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              22
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 gCFYag"
              width="73"
            >
              .718
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 dqgMn"
              width="59"
            >
              0
            </div>
          </div>
          <div
            className="am-flexbox am-flexbox-align-center"
            style={
              Object {
                "height": 30,
                "left": 0,
                "overflow": "hidden",
                "paddingRight": 0,
                "position": "absolute",
                "top": 30,
                "width": 300,
              }
            }
          >
            <div
              className="Table__Cell-sc-9u2sr5-2 fjWUKQ"
              style={
                Object {
                  "background": "#007a33",
                  "color": "#fff",
                  "fontWeight": "500",
                }
              }
              width="75"
            >
              BOS
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              53
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              25
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 gCFYag"
              width="73"
            >
              .679
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 dqgMn"
              width="59"
            >
              3
            </div>
          </div>
          <div
            className="am-flexbox am-flexbox-align-center"
            style={
              Object {
                "height": 30,
                "left": 0,
                "overflow": "hidden",
                "paddingRight": 0,
                "position": "absolute",
                "top": 60,
                "width": 300,
              }
            }
          >
            <div
              className="Table__Cell-sc-9u2sr5-2 fjWUKQ"
              style={
                Object {
                  "background": "#6f263d",
                  "color": "#fff",
                  "fontWeight": "500",
                }
              }
              width="75"
            >
              CLE
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              49
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              30
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 gCFYag"
              width="73"
            >
              .620
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 dqgMn"
              width="59"
            >
              7.5
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h3
      className="Table__Conference-sc-9u2sr5-3 iKGnKJ"
    >
      west
    </h3>
    <div
      className="ReactVirtualized__Table"
      role="grid"
      style={Object {}}
    >
      <div
        className="am-flexbox ReactVirtualized__Table__headerRow am-flexbox-align-center"
        style={
          Object {
            "height": 32,
            "overflow": "hidden",
            "paddingRight": 0,
            "width": 300,
          }
        }
      >
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 jhHCnp"
          width="75"
        >
          Teams
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 iGaVpi"
          width="47"
        >
          W
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 iGaVpi"
          width="47"
        >
          L
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 eZeJyr"
          width="73"
        >
          WIN%
        </div>
        <div
          className="Table__HeaderCell-sc-9u2sr5-1 cBrKBo"
          width="59"
        >
          GB
        </div>
      </div>
      <div
        aria-label="grid"
        aria-readonly={true}
        className="ReactVirtualized__Grid ReactVirtualized__Table__Grid"
        onScroll={[Function]}
        role="rowgroup"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "boxSizing": "border-box",
            "direction": "ltr",
            "height": 268,
            "overflowX": "hidden",
            "overflowY": "hidden",
            "position": "relative",
            "width": 300,
            "willChange": "transform",
          }
        }
        tabIndex={0}
      >
        <div
          className="ReactVirtualized__Grid__innerScrollContainer"
          role="rowgroup"
          style={
            Object {
              "height": 90,
              "maxHeight": 90,
              "maxWidth": 300,
              "overflow": "hidden",
              "pointerEvents": "",
              "position": "relative",
              "width": "auto",
            }
          }
        >
          <div
            className="am-flexbox am-flexbox-align-center"
            style={
              Object {
                "height": 30,
                "left": 0,
                "overflow": "hidden",
                "paddingRight": 0,
                "position": "absolute",
                "top": 0,
                "width": 300,
              }
            }
          >
            <div
              className="Table__Cell-sc-9u2sr5-2 fjWUKQ"
              style={
                Object {
                  "background": "#ba0c2f",
                  "color": "#fff",
                  "fontWeight": "500",
                }
              }
              width="75"
            >
              HOU
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              64
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              15
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 gCFYag"
              width="73"
            >
              .810
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 dqgMn"
              width="59"
            >
              0
            </div>
          </div>
          <div
            className="am-flexbox am-flexbox-align-center"
            style={
              Object {
                "height": 30,
                "left": 0,
                "overflow": "hidden",
                "paddingRight": 0,
                "position": "absolute",
                "top": 30,
                "width": 300,
              }
            }
          >
            <div
              className="Table__Cell-sc-9u2sr5-2 fjWUKQ"
              style={
                Object {
                  "background": "#ffc72d",
                  "color": "#fff",
                  "fontWeight": "500",
                }
              }
              width="75"
            >
              GSW
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              57
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              22
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 gCFYag"
              width="73"
            >
              .722
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 dqgMn"
              width="59"
            >
              7
            </div>
          </div>
          <div
            className="am-flexbox am-flexbox-align-center"
            style={
              Object {
                "height": 30,
                "left": 0,
                "overflow": "hidden",
                "paddingRight": 0,
                "position": "absolute",
                "top": 60,
                "width": 300,
              }
            }
          >
            <div
              className="Table__Cell-sc-9u2sr5-2 fjWUKQ"
              style={
                Object {
                  "background": "#f0163a",
                  "color": "#fff",
                  "fontWeight": "500",
                }
              }
              width="75"
            >
              POR
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              48
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 kiTAKI"
              width="47"
            >
              31
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 gCFYag"
              width="73"
            >
              .608
            </div>
            <div
              className="Table__Cell-sc-9u2sr5-2 dqgMn"
              width="59"
            >
              16
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
